x/amm/types/swap_out_amt_given_in.go

```diff
@@ -335,3 +332,19 @@ func (p *Pool) SwapOutAmtGivenIn(
 	}
 	return oracleOutCoin, slippage, slippageAmount, weightBalanceBonus, nil
 }
+
+// TODO: Ideally we should have a single DS for accounted pool to avoid confusion
+// Or refactor/improve amm pool to use accounted pool
+// Task has been added
```

Comment:
@amityadav0 the problem is that if both AMM and accounted pool DS are within a single module then you will have a dependency both between Perpetual -> AMM (to access accounted pool) and AMM -> Perpetual (to update accounted pool) created a bi-directional pattern that will cause cyclic dependencies and other architectural issues.